cloudwatch/cloudwatch.go

```diff
@@ -645,6 +645,22 @@ func (output *OutputPlugin) putLogEvents(stream *logStream) error {
 				stream.nextSequenceToken = &parts[len(parts)-1]
 
 				return output.putLogEvents(stream)
+			} else if awsErr.Code() == cloudwatchlogs.ErrCodeResourceNotFoundException {
+				// a log group or a log stream should be re-created after it is deleted and then retry
+				logrus.Errorf("[cloudwatch %d] Encountered error %v; detailed information: %s\n", output.PluginInstanceID, awsErr, awsErr.Message())
+				if strings.Contains(awsErr.Message(), "group") {
+					if err := output.createLogGroup(&Event{group: stream.logGroupName}); err != nil {
+						logrus.Error(err)
+						return err
+					}
+				} else if strings.Contains(awsErr.Message(), "stream") {
```

Comment:
String matching an error for `stream` and `group` feels fragile. I'm not sure of a better way to to this atm, but I feel like we should be able to identify the error type to avoid a situation where `group` becomes `Group` in a future version of the API. Just thoughts.

Reply:
I think we can't do a deeper type check that what already exists here- the check for `ErrCodeResourceNotFoundException`. 

The SDK doesn't give us different error types for the stream and group not existing AFAIK. I suppose to be safe we could make the check be case insensitive, so that "group" and "Group" match

Reply:
There is already a type check for `ErrCodeResourceNotFoundException` - then it does a string match to see if it's for group or stream. Again, I don't know if there's a better way to do this. Seeing string match code on errors makes me uneasy is all.